Morph2/src/core/net/sf/morph2/Defaults.java

```diff
@@ -42,7 +43,10 @@
  * @since Jan 9, 2005
  */
 public abstract class Defaults {
-
+	
+	// making these singletons to save memory
+	public static final TextToNumberConverter TEXT_TO_NUMBER_CONVERTER = new TextToNumberConverter();
```

Comment:
Gosh, it's been so long since I've worked with any of this stuff.  I just checked and I see lots of notes in BaseTransformer about thread safety.  If used properly, the object is thread safe.  That perhaps isn't the best design, but it can at least work without error.